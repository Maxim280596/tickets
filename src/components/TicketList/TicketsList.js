import React, {useEffect}  from "react";
import { useDispatch} from 'react-redux'
import { connect } from 'react-redux';

import { Ticket } from "../Ticket/Ticket";
import {fetchId} from '../../store/idReducer';
import {fetchTickets} from '../../store/ticketsReducer';


 const TicketsList = (props) => {
  // const [searchId, setSearchId] = useState("");
  // const [tickets, setTickets] = useState([]);

//  console.log(state.state.idReducer.searchId.searchId, 'hi')

// console.log(props)
const dispatch = useDispatch();


 
 useEffect(()=>{
   
dispatch(fetchId())
 
  
 }, [])

console.log(props.state)
 

//  useEffect(()=>{
   
// dispatch(fetchTickets(props.searchId.searchId))
//  console.log(props.searchId)
//  }, [props.searchId])  

//  console.log(state.state.idReducer.searchId.searchId)
 console.log(props, 'dssdfsddsd')

//  useEffect(()=>{
//   dispatch(fetchTickets(state.state.idReducer.searchId.searchId))
// }, [state.state.idReducer.searchId.searchId])

//  useEffect(()=>{
//    if(state.state.reducer.searchId) {
//   dispatch(putTickets())
//    }
//  }, [state.state.reducer.searchId, dispatch])
 


// console.log(props.store.reducer.data.searchId)
    
     
    
      

  // console.log(tickets);


  // console.log(searchId);

  return (
    <div>
      {/* {tickets.map((item) => (
        <Ticket
          key={item.price.toString()}
          price={item.price}
          origin={item.segments[0].origin}
          destination={item.segments[0].destination}
          originBack={item.segments[1].origin}
          destinationBack = {item.segments[1].destination}
          stops = {item.segments[0].stops.length}
          stopsBack = {item.segments[1].stops.length}
          stopsCity = {item.segments[0].stops.map((i) => (
            i
          )).join(', ')}
          stopsCityBack = {item.segments[1].stops.map((it) => (
            it
          )).join(', ')}
        />
      ))} */}
      <Ticket />
       
      <Ticket />
      <Ticket />
      <Ticket />
      <Ticket />
      <Ticket />
    </div>
  );
}

const mapStateToProps = function(state) {
  return {
    state: state
    // loggedIn: state.auth.loggedIn
  }
}

export default connect(mapStateToProps)(TicketsList);


