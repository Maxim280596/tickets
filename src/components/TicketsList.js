import React, { useEffect } from 'react';
import { connect, useDispatch, useSelector } from 'react-redux';
import { v4 as uuidv4 } from 'uuid';

import { Ticket } from './Ticket';
import { fetchId } from '../src/redux/ducks/idReducer';
import { fetchTickets, renderTickets } from '../src/redux/ducks/ticketsReducer';

const TicketsList = ({ state }) => {
  const dispatch = useDispatch();
  const searchId = useSelector((state) => state.idReducer.searchId);
  const loading = useSelector((state) => state.ticketsReducer.loadingFinish);

  useEffect(() => {
    dispatch(fetchId());
  }, []);

  useEffect(() => {
    return () => {
      dispatch(fetchTickets());
    };
  }, [searchId, dispatch]);

  useEffect(() => {
    return () => {
      dispatch(renderTickets());
    };
  }, [loading, dispatch]);

  console.log(state.loadingFinish && state?.tickets[0].tickets);
  return (
    <div>
      {state.loadingFinish ? (
        state.renderTickets.map((item) => {
          console.log();
          return (
            <Ticket
              key={uuidv4()}
              price={item.price}
              origin={item.segments[0].origin}
              destination={item.segments[0].destination}
              originBack={item.segments[1].origin}
              destinationBack={item.segments[1].destination}
              stops={item.segments[0].stops.length}
              stopsBack={item.segments[1].stops.length}
              stopsCity={item.segments[0].stops.join()}
              stopsCityBack={item.segments[1].stops.join()}
              departureDate={item.segments[0].date}
              duration={item.segments[0].duration}
              departureDateBack={item.segments[1].date}
              durationBack={item.segments[1].duration}
            />
          );
        })
      ) : (
        <h1>Loading</h1>
      )}
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    state: state.ticketsReducer,
  };
};

export default connect(mapStateToProps, null)(TicketsList);
