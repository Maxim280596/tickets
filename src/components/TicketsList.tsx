import React, { useEffect } from 'react';
import { connect, useDispatch, useSelector, ConnectedProps } from 'react-redux';
import { v4 as uuidv4 } from 'uuid';

import { Ticket } from './Ticket';
import {
  asyncSendRequestAction,
  renderTicket,
} from '../redux/ducks/ticketsReduser';
import { TicketsMap,RootState, RootStates } from '../types';
interface Props extends PropsFromRedux {
  isLoaded: boolean;
  data: [];
  renderTickets: [];
  error: string;
}

const TicketsList = (props: Props) => {
  const dispatch = useDispatch();
  const { isLoaded, data } = useSelector((state: RootState) => state);

  useEffect(() => {
    dispatch(asyncSendRequestAction());
  }, [dispatch, data]);

  useEffect(() => {
    return () => {
      if (isLoaded && data) {
        dispatch(renderTicket());
      }
    };
  }, [isLoaded, dispatch, data]);
  console.log(props);

  return (
    <>
      {props.data ? (
        props.renderTickets.map(
          (item: TicketsMap) => {
            return (
              <Ticket
                key={uuidv4()}
                price={item.price}
                origin={item.segments[0].origin}
                destination={item.segments[0].destination}
                originBack={item.segments[1].origin}
                destinationBack={item.segments[1].destination}
                stops={item.segments[0].stops.length}
                stopsBack={item.segments[1].stops.length}
                stopsCity={item.segments[0].stops.join()}
                stopsCityBack={item.segments[1].stops.join()}
                departureDate={item.segments[0].date}
                duration={item.segments[0].duration}
                departureDateBack={item.segments[1].date}
                durationBack={item.segments[1].duration}
              />
            );
          }
        )
      ) : (
        <h2>{props.error}</h2>
      )}
    </>
  );
};

const mapState = (state: RootStates) => ({
  isLoaded: state.mainReducer.isLoaded,
  data: state.mainReducer.data,
  renderTickets: state.mainReducer.renderTickets,
  error: state.mainReducer.error,
});
const connector = connect(mapState);
type PropsFromRedux = ConnectedProps<typeof connector>;

export default connector(TicketsList);
