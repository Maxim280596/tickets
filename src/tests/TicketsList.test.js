import React from 'react';
import {render} from 'enzyme';
import toJson from 'enzyme-to-json';
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import TicketsList from '../components/TicketsList';

describe('Testing <TicketList/>', () => {
  const initialState = {data:[]}
  const mockStore = configureStore()
  let state = {data:[]}
  let store;
  it('TicketsList have rendered correctly', ()=>{
    store = mockStore(initialState)
    const ticketslist = render(<Provider store={store}><TicketsList state={state} /></Provider>)
    expect(toJson(ticketslist)).toMatchSnapshot();
  })
})