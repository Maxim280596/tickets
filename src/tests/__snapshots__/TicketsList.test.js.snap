// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <TicketList/> TicketsList have rendered correctly 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(TicketsList) />
</ContextProvider>
`;
